device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -572,6 +572,13 @@ public void onTransportError(Event event)
 
         this.savedException = AmqpsExceptionTranslator.convertFromAmqpException(errorCondition);
 
+        if (this.isMultiplexing)
+        {
+            // transport errors involve closing the entire amqp connection, so save all the previous sessions
+            // so that their twin/methods subscriptions can persist
+            this.reconnectingDeviceSessionHandlers.putAll(this.sessionHandlers);
+        }
```

Comment:
I caught this as an issue because the canary pipelines had some multiplexing client tests failing due to the service actually killing the TCP connection when you send a TCP fault injection message now. Previously, it would just close an AMQP link.

Reply:
With this change, our tests pass in both prod and canary regions